### ISSUE TITLE
Wrong generated query with WHERE constraint following a SELECT MANY statement

### ISSUE THREAD

**Issue by bpatra** [Azure/azure-cosmos-dotnet-v2#208]
It looks like the query builder of the DocumentDB .NET SDK has a bug when generating SELECT MANY with WHERE constraints

See the following model

```
public class Player
{
            public string Name { get; set; }
            public bool IsGood { get; set; }
            public double Skills { get; set; }
}

//root document
public class FootballClub
{
        public string Name { get; set; }
        public Player[] TeamMembers {get; set;}
}

public class ClubPlayer
{
        public string ClubName { get; set; }
        public Player Player { get; set; }
}
```

I am trying to retrieve the association `ClubPlayer` but only for the good ones so I create this query

```
var goodPlayers = client.CreateDocumentQuery<FootballClub>(uri)
                .SelectMany(c => c.TeamMembers.Select(f => new ClubPlayer() { ClubName = c.Name, Player = f }))
                .Where(f => f.Player.IsGood).AsQueryable().AsEnumerable().ToArray();
```

But the generated query is incorrect see the generated query 
```
{"query":"SELECT VALUE {\"ClubName\": root[\"Name\"], \"Player\": f} FROM root JOIN f IN root[\"TeamMembers\"] WHERE f[\"Player\"][\"IsGood\"] "}
```
the `Where` constraint is incorrect we should have `f[\"IsGood\"]`
 
I could filter within the `Where` constraint like this 
````
.SelectMany(c => c.TeamMembers.Where(f => f.IsGood).Select(f => new ClubPlayer() { ClubName = c.Name, Player = f }))`
````
But it prevents me from ordering or filtering with a combination of `Club` and `Player` values. 
For example now my query is a bit more complex, I want to orderby descending on player `Skills`

Due to the wrong generated query the following does not work

```
 var players = client.CreateDocumentQuery<FootballClub>(uri)
                .SelectMany(c => c.TeamMembers.Select(f => new ClubPlayer() { ClubName = c.Name, Player = f }))
                .Where(comb => comb.Player.IsGood)
                .OrderByDescending(c=>c.Player.Skills).AsQueryable().AsEnumerable().ToArray();
```
While this one throws the error `Order-by over correlated collections is not supported.`

```
 var players = client.CreateDocumentQuery<FootballClub>(uri)
                .SelectMany(c => c.TeamMembers.Where(f => f.IsGood).Select(f => new ClubPlayer() { ClubName = c.Name, Player = f }))
                .OrderByDescending(c=>c.Player.Skills).AsQueryable().AsEnumerable().ToArray();
throws the following exception
```
For my case which is similar to this one. I could not use the Linq query but a brut SQL string query.

**Comment by wtodd** [comment]
This really needs to be fixed.  Are there any updates at all on this?

**Comment by khdang** [comment]
The fix for this and #91 is under review. It will be released in the release after it's merged.

**Comment by khdang** [comment]
This has been fixed with the latest release 2.4.0.
